### PR TITLE
EWL-4192 Article Preview

### PR DESCRIPTION
**Pull request by designerneil**
## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Jira Ticket**

- [EWL-4192: Migrate article preview](https://issues.ama-assn.org/browse/EWL-4192)

## Description

Updates the article preview styling and sizing of video 

## To Test

- [ ] http://localhost:3000/?p=molecules-article-stub
- [ ] Observe an image with a description below it 
- [ ] http://localhost:3000/?p=molecules-article-stub-related
- [ ] Observe an image with and h2 and paragraph on the right hand side
- [ ] http://localhost:3000/?p=molecules-article-stub-small-image
- [ ] Observe a copy with an image on the right 
- [ ] http://localhost:3000/?p=molecules-article-stub-title-only
- [ ] The title only should only have the title and read and date
- [ ] http://localhost:3000/?p=molecules-article-stub-video
- [ ] Observe a video with a description below it

## Visual Regressions

does this match SG1? https://americanmedicalassociation.github.io/AMA-style-guide/?p=molecules-article-preview

## Relevant Screenshots/GIFs
screen shot 2018-01-24 at 12 22 48 pm

## Remaining Tasks

Remaining tasks?

## Additional Notes

Anything more to add? Good things to call out here are:
- are there any PRs in this or other repositories that relate to or affect this PR?
- are there any caveats or oddities testers should know about when testing this PR?
- are there any problems you ran into during your work that you'd like to call out?

### CODE REVIEW

File: styleguide/source/_patterns/02-molecules/article-stub.twig [hunk 1]
```diff
@@ -20,31 +20,30 @@
...
-  {% if readtime and date and not related %}
-    <div class="ama__metadata article-stub__metadata col-xs-12">
-      {% include "@molecules/article-metadata.twig" %}
-    </div>
+{% endmacro %}
+
+{% macro metadata(articleStub) %}
+  {% set metadata = articleStub.metadata %}
+  {% if metadata and not title and not related %}
+    {% include "@molecules/article-metadata.twig" %}
```
Comment (labbydev) [inline styleguide/source/_patterns/02-molecules/article-stub.twig]: The metadata appears to be coming in on the `article-stub-related` and I don't believe that it should be there based on what I've seen in the News Article wireframe and the current Topics page.

screen shot 2018-01-24 at 4 13 59 pm

File: styleguide/source/assets/scss/02-molecules/_article-stub.scss [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    .ama__article-stub__copy {
+      display: flex;
+      flex-direction: column;
+      @extend .col-xs-9;
+      @include gutter($padding-left-half...);
+      padding-top: 0;
+    }
+  }
+
+  &--small {
```
Comment (labbydev) [inline styleguide/source/assets/scss/02-molecules/_article-stub.scss:39]: This is not floating the image appropriately. This is probably a good use case for floats as opposed to flexbox.

screen shot 2018-01-24 at 4 17 31 pm

File: styleguide/source/_patterns/02-molecules/article-stub.twig (hunk 1 above)
Comment (labbydev) [reply styleguide/source/_patterns/02-molecules/article-stub.twig]: Metadata is also appearing in the `article-stub-title-only` when it should not be there. 

screen shot 2018-01-24 at 4 20 09 pm

File: styleguide/source/_patterns/02-molecules/article-metadata.twig [hunk 3]
```diff
@@ -1,4 +1,2 @@
-<div class="ama__metadata ama__topic_article-preview_metadata">
-   <span>{{ date|default('Jul 28, 2017') }}</span> |
-   <span class="topic_article-preview_metadata_read-time">{{ readtime|default("5 min read") }}</span>
-</div>
+<span>{{ date|default('Jul 28, 2017') }}</span> |
```
Comment (labbydev) [inline styleguide/source/_patterns/02-molecules/article-metadata.twig:1]: Would there ever be a case where the date would appear but the time would not time and therefore we would not want the `|` to appear? Do we need an `if` statement here to account for that?

File: styleguide/source/assets/scss/01-atoms/_videos.scss [hunk 4]
```diff
@@ -9,3 +9,16 @@ figcaption {
   font-style: italic;
 }
 
+.ama__video-container {
```
Comment (labbydev) [inline styleguide/source/assets/scss/01-atoms/_videos.scss]: `container` here should be a sub element of `ama__video` and denoted with a double underscore.

File: styleguide/source/assets/scss/01-atoms/_videos.scss [hunk 5]
```diff
@@ -9,3 +9,16 @@ figcaption {
...
+  padding-top: 50%;
+
+  iframe {
+    width: 100%;
+    height: 100%;
+    position: absolute;
+    top: 0;
+    left: 0;
+  }
+}
```
Comment (labbydev) [inline styleguide/source/assets/scss/01-atoms/_videos.scss]: empty line

File: styleguide/source/assets/scss/02-molecules/_article-metadata.scss [hunk 6]
```diff
@@ -1,12 +1,8 @@
...
   display: block;
   margin-left: 0;
   margin-top: 14px;
   font-size: 12px;
   text-transform: uppercase;
-}
-
-.ama__topic_article-preview_metadata_read-time {
-  text-transform: lowercase;
 }
```
Comment (labbydev) [inline styleguide/source/assets/scss/02-molecules/_article-metadata.scss]: empty line

File: styleguide/source/assets/scss/02-molecules/_article-stub.scss [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  &--related {
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: row;
+    align-items: center;
+    @include ama-rules(1px, '', solid, $gray-50);
+    @include gutter($padding-top-half...);
+    @include gutter($padding-bottom-half...);
```
Comment (labbydev) [inline styleguide/source/assets/scss/02-molecules/_article-stub.scss]: includes should come before other items in scss

File: styleguide/source/assets/scss/02-molecules/_article-stub.scss [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        align-items: center;
+      }
+    }
+
+    .ama__article-stub__copy {
+      display: flex;
+      flex-direction: column;
+      @extend .col-xs-9;
+      @include gutter($padding-left-half...);
+      padding-top: 0;
```
Comment (labbydev) [inline styleguide/source/assets/scss/02-molecules/_article-stub.scss]: includes and extends should come before other rules in scss

**Review by labbydev — CHANGES_REQUESTED** [review]
I didn't see any of the `json` files for the variants here but the variant names should be "Article Stub as Variant" I'm seeing some mild discrepancies in the menu that should be remediated.

File: styleguide/source/_patterns/02-molecules/article-metadata.twig (hunk 3 above)
Comment (designerneil) [reply styleguide/source/_patterns/02-molecules/article-metadata.twig:1]: This follows the SG1 pattern. I don't know if time would appear without a date

File: styleguide/source/_patterns/02-molecules/article-stub.twig (hunk 1 above)
Comment (designerneil) [reply styleguide/source/_patterns/02-molecules/article-stub.twig]: yeah. that was an oversight. will fix

File: styleguide/source/assets/scss/01-atoms/_videos.scss (hunk 4 above)
Comment (designerneil) [reply styleguide/source/assets/scss/01-atoms/_videos.scss]: great catch

File: styleguide/source/assets/scss/01-atoms/_videos.scss (hunk 5 above)
Comment (designerneil) [reply styleguide/source/assets/scss/01-atoms/_videos.scss]: booooo. hate this empty line thing

File: styleguide/source/assets/scss/02-molecules/_article-stub.scss (hunk 2 above)
Comment (designerneil) [reply styleguide/source/assets/scss/02-molecules/_article-stub.scss:39]: the content just needs flex: 1

Comment (designerneil) [reply styleguide/source/assets/scss/02-molecules/_article-stub.scss:39]: fixed

**Pull request merged**